src/orchestrator/orchestrator.rs

```diff
@@ -0,0 +1,281 @@
+use futures::future::join_all;
+use std::{cmp::min, error::Error};
+use tokio::sync::mpsc;
+use tokio_util::sync::CancellationToken;
+
+use super::entry_csv_parser::entry_parser;
+use crate::aggregation_merkle_sum_tree::AggregationMerkleSumTree;
+use crate::executor::ExecutorSpawner;
+
+pub struct Orchestrator<const N_ASSETS: usize, const N_BYTES: usize> {
+    executor_spawner: Box<dyn ExecutorSpawner>,
+    entry_csvs: Vec<String>,
+}
+
+impl<const N_ASSETS: usize, const N_BYTES: usize> Orchestrator<N_ASSETS, N_BYTES> {
+    fn new(executor_spawner: Box<dyn ExecutorSpawner>, entry_csvs: Vec<String>) -> Self {
+        Self {
+            executor_spawner,
+            entry_csvs,
+        }
+    }
+
+    // Calculate the range of tasks to be assigned to a executor.
+    //
+    // * `executor_index` - The index of the executor.
+    // * `total_executors` - The total number of executor.
+    //
+    // A tuple representing the start and end indices of the tasks assigned to the executor
+    fn calculate_task_range(
+        &self,
+        executor_index: usize,
+        total_executors: usize,
+    ) -> (usize, usize) {
+        let total_tasks = self.entry_csvs.len();
+        let base_tasks_per_executor = total_tasks / total_executors;
+        let extra_tasks = total_tasks % total_executors;
+
+        let start = executor_index * base_tasks_per_executor + min(executor_index, extra_tasks);
+        let end =
+            (executor_index + 1) * base_tasks_per_executor + min(executor_index + 1, extra_tasks);
+
+        (start, min(end, total_tasks))
+    }
+
+    /// Processes a list of CSV files concurrently using executors and aggregates the results.
+    ///
+    /// * `executor_count` - The number of executors to use.
+    ///
+    /// Data flow
+    ///
+    /// 1. Splits the list of CSV files into segments based on the number of available executors.
+    /// 2. A distribution thread loads each CSV file, parses it into `entries`, and sends these to `entries_tx`.
+    /// 3. Each executor receives `entries` from `entries_rx`, requests tasks to Worker, and sends results back through `tree_tx`.
+    /// 4. The processed data from all executors, collected from `tree_rx`, is aggregated into an `AggregationMerkleSumTree`.
+    /// 6. After processing, executors are terminated to release resources.
+    ///
+    async fn create_aggregation_mst(
+        self,
+        executor_count: usize,
+    ) -> Result<AggregationMerkleSumTree<N_ASSETS, N_BYTES>, Box<dyn Error>>
+    where
+        [usize; N_ASSETS + 1]: Sized,
+        [usize; 2 * (1 + N_ASSETS)]: Sized,
+    {
+        let entries_per_executor = self.entry_csvs.len() / executor_count;
+
+        let mut executors = Vec::new();
+        let mut result_collectors = Vec::new();
+
+        let channel_size = std::env::var("CHANNEL_SIZE")
```

Comment:
You can easily set a variable with a command like `CHANNEL_SIZE=6 cargo run --release`.
I didn't mention this in the `README` because I wasn't sure about the performance impact related to the buffer size of channels.